src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp

```diff
@@ -314,141 +314,105 @@ inline double less_or_equal(double a, double b) {
     }
 
-    auto max_type_expected = std::numeric_limits<ExpectedT>::max();
-    auto max_type_actual = std::numeric_limits<ActualT>::max();
-    auto min_type_expected = std::numeric_limits<ExpectedT>::min();
-    auto min_type_actual = std::numeric_limits<ActualT>::min();
-    Error abs_error(abs_threshold), rel_error(rel_threshold);
+    size_t shape_size_cnt = shape_size(expected_shape);
+    Error error(abs_threshold, rel_threshold);
+    const auto expected_data = expected.data<ExpectedT>();
+    const auto actual_data = actual.data<ActualT>();
     for (size_t i = 0; i < shape_size_cnt; ++i) {
         double expected_value = expected_data[i];
         double actual_value = actual_data[i];
-        if ((std::isinf(expected_value) || expected_value >= max_type_expected) &&
-            (std::isinf(actual_value) || actual_value >= max_type_actual)) {
-            continue;
-        } else if ((std::isinf(expected_value) || expected_value <= min_type_expected) &&
-                   (std::isinf(actual_value) || actual_value <= min_type_actual)) {
-            continue;
-        }
-        if (std::isnan(expected_value) && std::isnan(actual_value))
+        if (!check_value(expected_value) && !check_value(actual_value)) {
             continue;
-        if (std::isnan(expected_value)) {
-            std::ostringstream out_stream;
-            out_stream << "Expected value is NAN but Actual value is not on coordinate: " << i;
-            throw std::runtime_error(out_stream.str());
-        }
-        if (std::isnan(actual_value)) {
-            std::ostringstream out_stream;
-            out_stream << "Actual value is NAN but Expected value is not on coordinate: " << i;
-            throw std::runtime_error(out_stream.str());
         }
-
-        double abs = std::fabs(expected_value - actual_value);
-        double rel = expected_value ? (abs / std::fabs(expected_value)) : abs;
-        abs_error.update(abs, i);
-        rel_error.update(rel, i);
-    }
-    abs_error.mean /= shape_size_cnt;
-    rel_error.mean /= shape_size_cnt;
-
-    if (!(less_or_equal(abs_error.max, abs_threshold) && less_or_equal(rel_error.max, rel_threshold))) {
-        std::ostringstream out_stream;
-        out_stream << "abs_max < abs_threshold && rel_max < rel_threshold"
-                   << "\n\t abs_max: " << abs_error.max << "\n\t\t coordinate " << abs_error.max_coordinate
-                   << "; abs errors count " << abs_error.count << "; abs mean " << abs_error.mean << "; abs threshold "
-                   << abs_threshold << "\n\t rel_max: " << rel_error.max << "\n\t\t coordinate "
-                   << rel_error.max_coordinate << "; rel errors count " << rel_error.count << "; rel mean "
-                   << rel_error.mean << "; rel threshold " << rel_threshold;
-        throw std::runtime_error(out_stream.str());
+        error.update(expected_value, actual_value, i);
     }
+    error.get_results();
```

Comment:
```suggestion
    error.check_results();
```

`get` in general means it return some value, so looks a bit confusing

Reply:
done